loader/cJSON.c

```diff
@@ -1215,3 +1222,133 @@ void cJSON_Minify(char *json) {
+    }
+    // NOTE: We can't just use fseek(file, 0, SEEK_END) because that isn't guaranteed to be supported on all systems
+    size_t fread_ret_count = 0;
+    do {
+        char buffer[256];
+        fread_ret_count = fread(buffer, 1, 256, file);
+    } while (fread_ret_count == 256 && !feof(file));
+    len = ftell(file);
+    fseek(file, 0, SEEK_SET);
+    json_buf = (char *)loader_instance_heap_calloc(inst, len + 1, VK_SYSTEM_ALLOCATION_SCOPE_COMMAND);
+    if (json_buf == NULL) {
+        loader_log(inst, VULKAN_LOADER_ERROR_BIT, 0,
+                   "loader_get_json: Failed to allocate space for JSON file %s buffer of length %d", filename, len);
+        res = VK_ERROR_OUT_OF_HOST_MEMORY;
+        goto out;
+    }
+    if (fread(json_buf, sizeof(char), len, file) != len) {
+        loader_log(inst, VULKAN_LOADER_ERROR_BIT, 0, "loader_get_json: Failed to read JSON file %s.", filename);
+        res = VK_ERROR_INITIALIZATION_FAILED;
+        goto out;
+    }
+    json_buf[len] = '\0';
+
+    // Can't be a valid json if the string is of length zero
+    if (len == 0) {
+        res = VK_ERROR_INITIALIZATION_FAILED;
+        goto out;
+    }
+    // Parse text from file
+    *json = cJSON_Parse(inst ? &inst->alloc_callbacks : NULL, json_buf);
+    if (*json == NULL) {
+        loader_log(inst, VULKAN_LOADER_ERROR_BIT, 0,
+                   "loader_get_json: Failed to parse JSON file %s, this is usually because something ran out of memory.", filename);
+        res = VK_ERROR_OUT_OF_HOST_MEMORY;
+        goto out;
+    }
+
+out:
+    loader_instance_heap_free(inst, json_buf);
+    if (NULL != file) {
+        fclose(file);
+    }
+
+    return res;
+}
+
+VkResult loader_parse_json_string_to_existing_str(const struct loader_instance *inst, cJSON *object, const char *key,
+                                                  size_t out_str_len, char *out_string) {
+    cJSON *item = cJSON_GetObjectItem(object, key);
+    if (NULL == item) return VK_ERROR_INITIALIZATION_FAILED;
+
+    char *str = cJSON_Print(item);
+    if (str == NULL) return VK_ERROR_OUT_OF_HOST_MEMORY;
+    strncpy(out_string, str, out_str_len);
+    out_string[out_str_len - 1] = '\0';
+    loader_instance_heap_free(inst, str);
+    return VK_SUCCESS;
+}
+
+VkResult loader_parse_json_string(const struct loader_instance *inst, cJSON *object, const char *key, char **out_string) {
+    cJSON *item = cJSON_GetObjectItem(object, key);
+    if (NULL == item) return VK_ERROR_INITIALIZATION_FAILED;
+
+    char *str = cJSON_Print(item);
+    if (str == NULL) return VK_ERROR_OUT_OF_HOST_MEMORY;
+    *out_string = str;
+    return VK_SUCCESS;
+}
+VkResult loader_parse_json_array_of_strings(const struct loader_instance *inst, cJSON *object, const char *key,
+                                            struct loader_string_list *string_list) {
+    VkResult res = VK_SUCCESS;
+    cJSON *item = cJSON_GetObjectItem(object, key);
+    if (NULL == item) return VK_ERROR_INITIALIZATION_FAILED;
```

Comment:
Inconsistent braces for if statements. I don't have a preference. But it's best to be consistent within a function.

Reply:
Yes, all if statements really should use braces to be consistent.